### DISCUSSION TITLE
Missing visibility for VaporImage

### DISCUSSION THREAD

**Discussion by riabiy-denis** [laravel/nova-issues#5265] [Ideas & Feature Requests]
### Description

It's not possible to set visibility on-demand for the file you upload. 

**pls, don't auto-close without reading further. I tried to be as short and to the point, as possible**

### Broader explanation/question

Do you plan to give an ability to pass a visibility to `VaporImage`? 
Example from [Vapor docs](https://docs.vapor.build/1.0/resources/storage.html#file-uploads).

```js
Vapor.store(this.$refs.file.files[0], {
    visibility: 'public-read'
}).then(response => {
    // ...
});
```

Would be nice just to `VaporImage::make('Avatar')->visibility('public-read')`.

### V1
- create custom field
- extend `VaporFileField`
- change `uploadVaporFiles` method to a custom one

### V2
Tried [this solution](https://github.com/laravel/nova-issues/discussions/3186), but it didn't work for some reason

```php
<?php

//...
class PublicVaporImage extends VaporImage
{
    protected function storeFile($request, $requestAttribute)
    {
        $filePath = parent::storeFile($request, $requestAttribute);

        $disk = Storage::disk($this->getStorageDisk());
        $disk->setVisibility(rtrim($filePath, '/'), 'public');

        return $filePath;
    }
}
```

### Question/outcome

Could you tell me, if it's smth you plan to do at all? Cuz if yes -> I'm in the position, when I can simply wait for 1-2mo and assume it will be there. From your side it seems like a really easy change in one of the fields. 

If you don't plan to do that at all, pls, just let me know and I'll go through the whole hustle of creating a custom field to make it work for me : )

Thank you a lot for reading so far. Hope to see your answer 🤗

**Comment by crynobone** [comment]
1. `Vapor.store()` only used to upload the file temporary directory and visibility should just follow the default.
2. At the moment, we only use the global options and adding this new feature should be marked as feature request.

**Reply by riabiy-denis** [reply]
@crynobone 
1. Thank you for reading.
2. That's why I've asked. It could be in your backlog...or not. Or smth you think "hmm...sounds like a nice idea".


> Vapor.store() only used to upload the file temporary directory and visibility should just follow the default.

It gets carried over to the copy too, when you do that. 
[YouTube video](https://www.youtube.com/watch?v=iKjpy3USuXs) from the Laravel channel with an explanation

And if you don't want to watch for 15 mins, the best timestamp I can give you is "from 11:36". 

In a nutshell, front sets the visibility, that gets carried over.
